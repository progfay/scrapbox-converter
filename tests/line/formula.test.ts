/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../../src'
import pretty from 'pretty'

describe('formula', () => {
  it('Simple formula', () => {
    const input = `Formula
[$ \sum_{k=0}^N k^2 = \frac{1}{2}N(N + 1)]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
