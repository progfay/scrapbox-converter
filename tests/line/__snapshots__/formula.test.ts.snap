// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formula Simple formula 1`] = `
"<article>
  <h1 class=\\"title\\">Formula</h1>
  <section>
    <p><span><span class=\\"katex-error\\" title=\\"ParseError: KaTeX parse error: Unexpected character: &#x27;&#x27; at position 19: …_{k=0}^N k^2 = ̲rac{1}{2}N(N + …\\" style=\\"color:#cc0000\\">sum_{k=0}^N k^2 = rac{1}{2}N(N + 1)</span></span></p>
  </section>
</article>"
`;
